File: MoviesApplication/Data/Models/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace MoviesApplication.Data.Models
{
    public class Movie
    {
        [Key]
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public bool? IsRecommended { get; set; }
        public DateTime? ReleaseDate { get; set; }
        public string? Genres { get; set; }
        public string? RunTime { get; set; }
        public int? AgeRating { get; set; }
        public string? MovieUrl { get; set; }
        public string? TrailerUrl { get; set; }
        public string? PosterUrl { get; set; }
        public int Likes { get; set; } = 0;
        public int Dislikes { get; set; } = 0;
        public ICollection<Director>? Directors { get; set; }
        public ICollection<Actor>? Actors { get; set; }
        

    }
}

File: MoviesApplication/ClientApp/src/app/shared/movie.ts
import Director from './director';
import { Actor } from './actor';

export class Movie {
  id!: string;
  title!: string;
  description!: string | null;
  isRecommended!: boolean | null;
  releaseDate!: Date | null;
  genres!: string | null;
  runTime!: string | null;
  ageRating!: number | null;
  movieUrl!: string | null;
  trailerUrl!: string | null;
  posterUrl!: string | null;
  directors!: Director[] | null;
  actors!: Actor[] | null;
  likes!: number;
  dislikes: number;
}
